### DISCUSSION TITLE
How to preserve the watched value on submit

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#5347] [Q&A] [answered]
I am using useWatch() to keep track of a total that is calculated from useFieldArray. This is all working fine except when I submit the form and there is a validation error (using Yup). Then the total goes back to 0 when it should stay at whatever it was. Is it possible to preserve the value on submit with failed validation?

```
  const {
    register,
    control,
    handleSubmit,
    formState: { errors },
  } = useForm({
    defaultValues,
    resolver: yupResolver(schema),
  });
  const { fields, append, remove } = useFieldArray({
    control,
    name: "items",
  });

  const PriceTotal = ({ control }) => {
    const value = useWatch({
      control,
      name: `items`,
      defaultValue: {},
    });

    let total;
    if (value && value.length) {
      total = value.reduce((prev, cur) => prev + cur.price * cur.qty, 0);
    } 
    return <span>total: {total || 0}</span>;
  };
```

Then down in the render where the form is... this is the value that goes to 0 on form submit if there are validation errors from Yup. 

`  <PriceTotal control={control} />`

**Comment by bluebill1049** [comment]
register an input and store that value, or simply use a input type="hidden"

**Comment by thanh-nguyen-95** [comment, marked as answer]
Have you tried the solution I suggested in your old thread #5316?

**Comment by ghost** [comment]
I did, but I kept getting undefined for the total value. I will post the code later today and hopefully you will be able to spot my mistake.

**Reply by thanh-nguyen-95** [reply]
You might have forgot to return the calculated value

```js
const total = useMemo(() => {
  const result = /* calculate here */

  return result
}, [value])
```

**Comment by ghost** [comment]
That actually looks like it could be exactly the problem as I didn’t return it as far as I can remember. Thanks so much, will give it a go when I get a gap later and let you know. Thanks again!

**Comment by ghost** [comment]
This works now, thanks!! :)
